File: app/src/main/java/com/mouredev/weeklychallenge2022/Challenge38.php
<?php

declare(strict_types=1);

/*
 * Reto #38
 * BINARIO A DECIMAL
 * Fecha publicación enunciado: 19/09/22
 * Fecha publicación resolución: 27/09/22
 * Dificultad: MEDIA
 *
 * Enunciado: Crea un programa se encargue de transformar un número binario a decimal sin utilizar
 * funciones propias del lenguaje que lo hagan directamente.
 *
 * Información adicional:
 * - Usa el canal de nuestro Discord (https://mouredev.com/discord) "🔁reto-semanal"
 *   para preguntas, dudas o prestar ayuda a la comunidad.
 * - Tienes toda la información sobre los retos semanales en
 *   https://retosdeprogramacion.com/semanales2022.
 *
 */

$binary = php_sapi_name() === 'cli'
    ? ($argv[1] ?? false)
    : ($_GET['binary'] ?? false);

$binary = (string) ($binary ?: 10111); // 23

try {

    $decimal = convertBinaryToDecimal($binary);

    echo "The decimal of \"{$binary}\" is: {$decimal}.";

} catch (Exception $e) {

    echo $e->getMessage();

}

echo PHP_EOL;

function convertBinaryToDecimal(string $binary): int
{
    $original = $binary;

    if (!is_numeric($binary)) {
        throw new Exception("The binary `{$original}` is not a valid binary");
    }

    $binary = str_split($binary);
    $binary = array_reverse($binary);

    foreach ($binary as $index => &$number) {

        if (!in_array($number, [0, 1])) {
            throw new Exception("The binary `{$original}` is not a valid binary");
        }

        $number = $number * 2 ** $index;

    }

    $decimal = (int) array_sum($binary);

    return $decimal;
}

File: app/src/main/cpp/Challenge38.cpp
/*
 * Reto #38
 * BINARIO A DECIMAL
 * Fecha publicación enunciado: 19/09/22
 * Fecha publicación resolución: 27/09/22
 * Dificultad: MEDIA
 *
 * Enunciado: Crea un programa se encargue de transformar un número binario a decimal sin utilizar
 * funciones propias del lenguaje que lo hagan directamente.
 *
 * Información adicional:
 * - Usa el canal de nuestro Discord (https://mouredev.com/discord) "🔁reto-semanal"
 *   para preguntas, dudas o prestar ayuda a la comunidad.
 * - Tienes toda la información sobre los retos semanales en
 *   https://retosdeprogramacion.com/semanales2022.
 *
 */

#include <iostream>
#include <math.h>
using namespace std;

int main(int argc, char const *argv[])
{

    int decimal = 0;

    if (argc == 2)
    {
        string binNum = argv[1];
        int j = 0;
        for (int i = binNum.length() - 1; i >= 0; i--)
        {
            if (binNum[i] == '1')
            {
                decimal += pow(2, j);
            }
            j++;
        }
        cout << "El número binario: " << argv[1] << " equivale al numero decimal: " << decimal << endl;
    }
    else
    {
        cout << "Revisa el enunciado y vuelve a pasar los argumentos correctamente." << endl;
    }

    return 0;
}
